etc/helm/pachyderm/templates/pachd/config-secret.yaml

```diff
@@ -0,0 +1,104 @@
+{{- /*
+SPDX-FileCopyrightText: Pachyderm, Inc. <info@pachyderm.com>
+SPDX-License-Identifier: Apache-2.0
+*/ -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pachyderm-config
+  namespace: {{ .Release.Namespace }}
+stringData:
+{{- if eq .Values.deployTarget "LOCAL" }}
+  # authConfig is the configuration for pachd as an OIDC client
+  authConfig: |
+    client_id: pachd 
+    client_secret: oidcsecret
+    issuer: http://pachd:1658/
+    localhost_issuer: true
+    require_email_verified: false
+    redirect_uri: http://localhost:30657/authorization-code/callback
+    scopes:
+    - email
+    - profile
+    - groups
+    - openid 
+{{- end }}
+
+  # clusterRoleBindings is a set of user -> role mappings to apply
+  clusterRoleBindings: |
+    robot:test:
+    - repoReader
+
+  # enterpriseClusters is the set of pachds covered by license service 
+  enterpriseClusters: |
+    - address: grpc://localhost:1650
+      id: localhost
+      secret: secret
+      user_address: grpc://localhost:30650
+      cluster_deployment_id: {{ include "pachyderm.clusterDeploymentId" . }}
+
+  # enterpiseConfig points the pachd to a license service (in this case itself)
+{{- if eq .Values.deployTarget "LOCAL" }}
+  enterpriseConfig: |
+    id: localhost
+    license_server: grpc://localhost:1650
+    secret: secret
+{{- end }}
+
+  # identityServiceConfig configures the OIDC provider
+{{- if eq .Values.deployTarget "LOCAL" }}
+  # id_token_expiry value is parsed into golang time.Duration: https://pkg.go.dev/time#example-ParseDuration
+  identityServiceConfig: |  
+    issuer: http://pachd:1658/
+    id_token_expiry: 24h
+{{- end }}
+
+{{- if eq .Values.deployTarget "LOCAL" }}
+  # idps is the set of Identity Providers to support for logging in
+  idps: |
+    - id: test
+      name: test
+      type: mockPassword
+      jsonConfig: '{"username": "admin", "password": "password"}' 
+      
+  # idps: |
+  #   - id: okta
+  #     jsonConfig: >- 
+  #       {
+  #           "issuer": "https://dev-84362674.okta.com",
+  #           "clientID": "client_id",
+  #           "clientSecret": "notsecret",
+  #           "redirectURI": "http://localhost:30658/callback",
+  #           "insecureEnableGroups": true,
+  #           "insecureSkipEmailVerified": true,
+  #           "insecureSkipIssuerCallbackDomainCheck": true
+  #       }
+  #     name: okta
+  #     type: oidc
+{{- end }}
+
+  license: {{ required "Enterprise license key required" .Values.pachd.enterpriseLicenseKey | quote }}
+
+  enterpriseSecret: "secret" # {{ randAlphaNum 32 | quote }}
```

Comment:
We'll probably want to treat this similarly to clusterDeploymentId (option to pass in or generate value?)